docs/index.bs

```diff
@@ -1809,42 +1809,48 @@ spec: webappsec-referrer-policy; urlPrefix: https://w3c.github.io/webappsec-refe
+
+        1. Let |request| be null.
+        1. If |requestArg| is a {{Request}} object, then:
+            1. Set |request| to |requestArg|'s [=Request/request=].
+            1. If |request|'s [=request/url=]'s [=url/scheme=] is not one of "`http`" and "`https`", or |request|'s [=request/method=] is not \`<code>GET</code>\`, return [=a promise rejected with=] a `TypeError`.
+        1. Else if |requestArg| is a string, then:
+            1. Set |request| to the associated [=Request/request=] of the result of invoking the initial value of {{Request}} as constructor with |requestArg| as its argument. If this [=throws=] an exception, return [=a promise rejected with=] that exception.
+            1. If |request|'s [=request/url=]'s [=url/scheme=] is not one of "`http`" and "`https`", return [=a promise rejected with=] a `TypeError`.
+        1. Let |response| be |responseObject|'s [=Response/response=].
+        1. If |response|'s [=response/status=] is `206`, return [=a promise rejected with=] a `TypeError`.
+        1. If |response|'s [=response/header list=] contains a [=header=] [=header/named=] \`<code>Vary</code>\`, then:
             1. Let |fieldValues| be the [=list=] containing the [=list/items=] corresponding to the  [=Vary=] header's [=http/field-values=].
             1. [=list/For each=] |fieldValue| in |fieldValues|:
                 1. If |fieldValue| matches "`*`", return [=a promise rejected with=] a `TypeError`.
-        1. If |response| is [=Body/disturbed=] or [=Body/locked=], return [=a promise rejected with=] a `TypeError`.
-        1. Let |clonedResponse| be the result of [=response/clone|cloning=] |response|'s associated [=Response/response=].
-        1. If |response|'s body is non-null, run these substeps:
-            1. Let |dummyStream| be an [=empty=] {{ReadableStream}} object.
-            1. Set |response|'s [=response/body=] to a new [=/body=] whose [=stream=] is |dummyStream|.
-            1. Let |reader| be the result of [=get a reader|getting a reader=] from |dummyStream|.
-            1. [=Read all bytes=] from |dummyStream| with |reader|.
+        1. If |response|'s [=response/body=] is [=Body/disturbed=] or [=Body/locked=], return [=a promise rejected with=] a `TypeError`.
+        1. Let |clonedResponse| be a [=response/clone=] of |response|.
+        1. Let |bodyReadPromise| be [=a promise resolved with=] undefined.
+        1. If |response|'s [=response/body=] is non-null, run these substeps:
+            1. Let |stream| be |response|'s [=response/body=]'s [=body/stream=].
+            1. Let |reader| be the result of [=ReadableStream/get a reader|getting a reader=] for |stream|.
+            1. Set |bodyReadPromise| to the result of [=Read all bytes|reading all bytes=] from |stream| with |reader|.
+
+            Note: This ensures that |response|'s [=response/body=] is [=Body/locked=], and we have a full buffered copy of the body in |clonedResponse|.
```

Comment:
I don't understand this. It seems there shouldn't be a clone at all. And if the stream is already tainted you'd throw.

Reply:
I'm creating a clone so the algorithm has a non-consumed copy that can be stored.

We're already throwing if the stream is tainted, but this algorithm needs to taint the stream.

Reply:
At the very least I think there should be a note indicating that this is there to simulate reading the body in order to write it to disk.  We don't actually require tee'ing the data in memory before writing the bits on to disk.

Reply:
Interesting. What happens if the stream is rejected or a signal is aborted?

It also seems good to indicate that any failures are already dealt with for the original stream so this new stream should stream without error to disk assuming no hardware issues.

Reply:
@annevk `|bodyReadPromise|` is returned at the end of this algorithm, so any rejection there will become a rejection of `cache.put()`.

@wanderview good point. I'll expand on that.

Reply:
How does the user of the API observe cacheJobPromise in that case?

Reply:
`cacheJobPromise` is returned in the fulfillment handler of `bodyReadPromise`

Reply:
But the original version of this API returned cacheJobPromise, no?
```
bodyReadPromise.then(x => cacheJobPromise);
return bodyReadPromise
```
is not equal to
```
return cacheJobPromise
```
unless I'm missing something.

Reply:
oh ffs you're right. Fixing…

Reply:
(my excuse is jetlag)